src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -1159,6 +1162,11 @@ bool Convolution::isPossibleToSkipInitConfig(const dnnl::primitive_desc &desc) c
 }
 
 std::shared_ptr<MemoryDesc> Convolution::getSrcMemDesc(dnnl::primitive_desc_iterator &primitive_desc_it, size_t idx) {
+    if (idx == 1) {
+        // report original plain layout for weight since it needs to be reordered dynamically at runtime
+        return std::make_shared<CpuBlockedMemoryDesc>(getOriginalInputPrecisionAtPort(idx),
+                                                      Shape(getInputShapeAtPort(idx).getStaticDims()));
```

Comment:
Shouldn't we extract the precision from the actual weights? In my understanding only the layout should be replaced.

Reply:
In some model, the weight is provided in s32 or other precision which is un-supported by oneDNN, and CPU node will use f32 as inference precision in such case, so oneDNN primitive also requires f32 for weight precision.

if we report `f32+plain_layout` here, CPU graph will insert reorders which converts `s32+plain_layout` into `f32+plain_layout`, this only performed once for constant weight, but it only produces a temporary memory format, since after `prepareParam` stage, the layout is going to change again, so here I report the original precision specified in ngraph, to prevent the extra reorder nodes and temporary memories, one reorder in prepareWeightMemory() or DnnlExecutor() should be enough in all cases, right?

Reply:
That does make sense. Agree. This is another reason to refactor how we work with oneDNN.